haystack/pipelines/base.py

```diff
@@ -410,7 +411,14 @@ class Pipeline(BasePipeline):
     def __init__(self):
         self.graph = DiGraph()
         self.root_node = None
-        self.components: dict = {}
+
+    @property
+    def components(self):
```

Comment:
Minor naming question: why `components` and not `nodes`? Users aren't aware of `networkx`'s naming.

Reply:
We use both notations in pipeline config: components define the "parts" of the pipeline as such. Nodes define the interplay of these components. Semantically components is the better name here.